tests/test_core.py

```diff
@@ -2419,6 +2419,16 @@ def test_pthread_specific(self):
   def test_pthread_equal(self):
     self.do_run_in_out_file_test('pthread/test_pthread_equal.cpp')
 
+  @node_pthreads
+  def test_pthread_proxying(self):
+    self.set_setting('EXIT_RUNTIME')
+    self.set_setting('PROXY_TO_PTHREAD')
+    self.set_setting('PTHREAD_POOL_SIZE=3')
+    self.set_setting('INITIAL_MEMORY=32mb')
```

Comment:
Do you need 32mb here?

Reply:
I need something larger than the default, and I arbitrarily chose 32mb as something that worked. It seemed fishy that the test was using so much memory, but instrumenting `dlmalloc` to print allocation sizes and stack traces didn't produce anything unexpected. By far the largest allocations are for the the pthread stacks, but those are all properly freed once the threads are joined AFAICT.

Reply:
Out default pthread stack size is 2mb, which is way to large IMHO, but also you only have 3 threads here so I can't see how that would require this much memory.   lgtm either to this though.. trying to reduce memory overhead is not what this change is about.